src/test/synapse.test.ts

```diff
@@ -255,6 +259,138 @@ describe('Synapse', () => {
+      status: '0x1',
+      blockNumber: '0x127001',
+      cumulativeGasUsed: '0x52080',
+      gasUsed: '0x5208',
+      logs: [makeDataSetCreatedLog(DATA_SET_ID, 1)],
+    }
+    beforeEach(() => {
+      server.use(PING())
+      const pdpOptions: PDPMockOptions = {
+        baseUrl: 'https://pdp.example.com',
+      }
+      server.use(createDataSetHandler(FAKE_TX_HASH, pdpOptions))
+      server.use(
+        dataSetCreationStatusHandler(
+          FAKE_TX_HASH,
+          {
+            ok: true,
+            dataSetId: DATA_SET_ID,
+            createMessageHash: '',
+            dataSetCreated: true,
+            service: '',
+            txStatus: '',
+          },
+          pdpOptions
+        )
+      )
+    })
+
+    it('should storage.createContext with session key', async () => {
+      const signerAddress = await signer.getAddress()
+      const sessionKeySigner = new ethers.Wallet(PRIVATE_KEYS.key2, provider)
+      const sessionKeyAddress = await sessionKeySigner.getAddress()
+      const EXPIRY = BigInt(1757618883)
+      const FAKE_SIGNATURE =
+        '0x4355c47d63924e8a72e509b65029052eb6c299d53a04e167c5775fd466751c9d07299936d304c153f6443dfa05f40ff007d72911b6f72307f996231605b915621c'
+      server.use(
+        JSONRPC({
+          ...presets.basic,
+          sessionKeyRegistry: {
+            authorizationExpiry: (args) => {
+              const client = args[0]
+              const signer = args[1]
+              assert.equal(client, signerAddress)
+              assert.equal(signer, sessionKeyAddress)
+              const permission = args[2]
+              assert.isTrue(PDP_PERMISSIONS.includes(permission))
+              return [EXPIRY]
+            },
+          },
+          payments: {
+            ...presets.basic.payments,
+            operatorApprovals: (args) => {
+              const token = args[0]
+              const client = args[1]
+              const operator = args[2]
+              assert.equal(token, ADDRESSES.calibration.usdfcToken)
+              assert.equal(client, signerAddress)
+              assert.equal(operator, ADDRESSES.calibration.warmStorage)
+              return [
+                true, // isApproved
+                BigInt(127001 * 635000000), // rateAllowance
+                BigInt(127001 * 635000000), // lockupAllowance
+                BigInt(0), // rateUsage
+                BigInt(0), // lockupUsage
+                BigInt(28800), // maxLockupPeriod
+              ]
+            },
+            accounts: (args) => {
+              const token = args[0]
+              const user = args[1]
+              assert.equal(user, signerAddress)
+              assert.equal(token, ADDRESSES.calibration.usdfcToken)
+              return [BigInt(127001 * 635000000), BigInt(0), BigInt(0), BigInt(0)]
+            },
+          },
+          eth_signTypedData_v4: (params) => {
+            const user = params[0]
+            assert.equal(user, sessionKeyAddress)
```

Comment:
this looked right to me a few hours ago, but now that I think about it, signing typed data with a private key should not result in an API request like this. I'll look into it tomorrow.